Sort by address

pducolin opened the pull request:
Using https://koumoul.com/s/geocoder/api-doc to get missing positions, adding positions to cinemas.json to be able to sort them by distance from a position in frontend.

Commit:
move types declaration from clean-and-save to types

Commit:
add positionHelper file to sync cinemaPosition.json and cinemas.json
Export function do get missing positions using french open database (kumol.com)

Commit:
update cinePositions and add position to cinemas.json

Commit:
add position and zipcode to cinemas

kevinmiguet reviewed (approved):
Great job !

kevinmiguet commented on src/utils/positionHelper.ts:
```diff
@@ -0,0 +1,96 @@
+let cinemas: { [id: string]: Cinema } = require('../../cinemas.json');
+
+export async function getPositionForCine(cine: Cinema): Promise<acPosition> {
+    let pos: acPosition = positions.find((_cine: any) => _cine.address === cine.address )
```
might be safer to use the cine ID here, what do you think ?

pducolin replied:
yes, I should add it in the sync function

kevinmiguet commented on src/utils/positionHelper.ts:
```diff
@@ -0,0 +1,96 @@
+    if (!cine.zipCode) {
+        return null
+    }
+    try {
```
how about putting this in a dedicated "fetchAddress" function where we would always write the file when fetching a new address ?

pducolin replied:
I moved the sync part to a separate function, will move the async part too. Shall we push the positions file each time we update it?

kevinmiguet replied:
To be honest I'm not familiar with automatic commits with node, so I don't have a strong opinion about this : ) . However I'd say it's only a nice to have. If you think it's a good idea and really want it, feel free to do it !

Commit:
feat(positonHelper): get position for a cinema using cine id instead of cine address

Commit:
:twisted_rightwards_arrows: fix build after rebasing master in dev branch

Commit:
use command and query pattern
getters should not edit data, so I call fetch from the main function. Improving also perf using async write file thanks to promisify
